compose/service.py

```diff
@@ -1069,6 +1071,28 @@ def parse_restart_spec(restart_config):
 
     return {'Name': name, 'MaximumRetryCount': int(max_retry_count)}
 
+# Ulimits
+
+
+def build_ulimits(ulimit_config):
+    if not ulimit_config:
+        return None
+    ulimits = []
+    for limit_name, soft_hard_values in six.iteritems(ulimit_config):
+        if isinstance(soft_hard_values, six.integer_types):
+            ulimits.append({'name': limit_name, 'soft': soft_hard_values, 'hard': soft_hard_values})
+        elif isinstance(soft_hard_values, dict):
+            if not soft_hard_values['soft'] <= soft_hard_values['hard']:
+                raise ConfigError(
```

Comment:
Wait, hang on, I'm in a different file. Sorry.

The checking of these values should go in as a separate `validate_ulimits` function over in `config.py` and raise a `ConfigurationError`. We want to raise configuration errors early and make them the concern of the config not the service.